appsec/src/extension/ddappsec.c

```diff
@@ -451,6 +457,46 @@ static PHP_FUNCTION(datadog_appsec_testing_request_exec)
     RETURN_TRUE;
 }
 
+static PHP_FUNCTION(datadog_appsec_push_params)
+{
+    UNUSED(return_value);
+    if (!DDAPPSEC_G(active)) {
+        mlog(dd_log_debug, "Trying to access to push_params "
+                           "function while appsec is disabled");
+        return;
+    }
+
+    zval *parameters = NULL;
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "z", &parameters) == FAILURE ||
+        Z_TYPE_P(parameters) != IS_ARRAY) {
+        mlog(dd_log_warning, "Unexpected parameters type. Expected array");
+        return;
+    }
+
+    zval parameters_zv;
+    zend_array *parameters_arr = zend_new_array(1);
+    ZVAL_ARR(&parameters_zv, parameters_arr);
+    zval *res = zend_hash_add(
+        Z_ARRVAL(parameters_zv), _key_server_request_path_params, parameters);
+    if (res == NULL) {
+        zval_ptr_dtor(&parameters_zv);
+        mlog_g(dd_log_debug, "Parameters could not be added");
+        return;
+    }
+    Z_ADDREF_P(parameters);
+
+    dd_conn *conn = dd_helper_mgr_cur_conn();
+    if (conn == NULL) {
+        zval_ptr_dtor(&parameters_zv);
+        mlog_g(dd_log_debug, "No connection; skipping push_params");
+        return;
+    }
+
+    dd_request_exec(conn, &parameters_zv);
```

Comment:
`dd_request_exec` will provide you with the WAF verdict so you need to handle it and block / redirect accordingly.

Reply:
Done